apps/content-gateway-api/src/app/maintenance/jobs/index-handling/IndexCreationJob.ts

```diff
@@ -0,0 +1,146 @@
+import { pipe } from "fp-ts/lib/function";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import { MaintenanceJob, MaintenanceJobError } from "../..";
+import * as t from "io-ts";
+import { Db } from "mongodb";
+// @ts-expect-error There are no type definitions for this package :'(
+import * as DigestFetch from "digest-fetch";
+
+/**
+ * Queries MongoDB for index recommendations and then
+ * enacts those recommendations
+ */
+export const createIndexCreationJob = (
+    atlasApiInfo: AtlasApiInfo,
+    db: Db
+): MaintenanceJob => {
+    const runIndexCreation = (): TE.TaskEither<MaintenanceJobError, void> => {
+        return pipe(
+            _queryIndexSuggestions(atlasApiInfo),
+            TE.chain(_addIndexes(db))
+        );
+    };
+
+    return {
+        run: runIndexCreation,
+        id: "CreateIndexes",
+        schedule: {
+            hours: 1,
+        },
+    };
+};
+
+export interface AtlasApiInfo {
+    publicKey: string;
+    privateKey: string;
+    projectId: string;
+    processId: string;
+}
+
+/**
+ * INTERNAL OBJECTS, DO NOT USE:
+ * they are exported for testing purposes only
+ */
+
+/**io-ts codec for verifying external input*/
+const IndexSuggestionCodec = t.strict({
+    shapes: t.unknown, // We aren't using this so idc what it is
+    suggestedIndexes: t.array(
+        t.strict({
+            id: t.string,
+            impact: t.array(t.string),
+            index: t.array(t.record(t.string, t.number)),
+            namespace: t.string,
+            weight: t.number,
+        })
+    ),
+});
+
+export type IndexSuggestions = t.TypeOf<typeof IndexSuggestionCodec>;
+
+// TODO: remove hard coded process Id with automatic discovery of processes
```

Comment:
https://www.notion.so/bankless/Remove-hard-coded-process-Id-with-automatic-discovery-of-processes-f1aabbc08c9a40f9a9f7e6c1e2eafc06